### PR TITLE
Fix unit tests for non-ATLAS policies : Closes #5605

### LINKED ISSUE #5605: Fix VO specific tests
Motivation
----------
A  lot of tests are still disabled for VO-specific tests because of wrong naming convention for DIDs or RSEs. They need to be fixed

### PR DESCRIPTION
**Pull request by cserf**
Fix unit tests for non-ATLAS policies : Closes #5605

This PR starts fixing the unit tests that are failing for non-ATLAS policies (in particular belle II). Since many tests need a fix, only a few tests are fixed here (listed below):
```
 lib/rucio/tests/test_abacus_account.py
 lib/rucio/tests/test_abacus_collection_replica.py
 lib/rucio/tests/test_abacus_rse.py
 lib/rucio/tests/test_account.py
 lib/rucio/tests/test_api_external_representation.py
 lib/rucio/tests/test_did.py
 lib/rucio/tests/test_dirac.py
 lib/rucio/tests/test_permission.py
 lib/rucio/tests/test_reaper.py
 lib/rucio/tests/test_replica.py
 lib/rucio/tests/test_rse.py
 lib/rucio/tests/test_rule.py
 lib/rucio/tests/test_subscription.py
 lib/rucio/tests/test_undertaker.py
```

Some tests cannot be fixed because some of the features that they are testing only works for ATLAS. In that case they were disabled and an issue was open to fix the feature to make it work with other policies (e.g. https://github.com/rucio/rucio/issues/5612)

### CODE REVIEW

File: lib/rucio/tests/common.py [hunk 1]
```diff
@@ -66,13 +67,15 @@ def scope_name_generator():
...
-def did_name_generator(did_type='file'):
+def did_name_generator(did_type='file', name_prefix='', name_suffix='', cnt=0):
     """ Generate random did name.
 
     :returns: A random did name
     """
     if os.getenv('POLICY') == 'belleii':
-        path = '/belle'
+        path = '/belle/mock' + name_prefix
+        if not path[-1] != '/':
```
Comment (maany) [inline lib/rucio/tests/common.py]: umm, double negation?

File: lib/rucio/tests/common.py [hunk 2]
```diff
@@ -66,13 +67,15 @@ def scope_name_generator():
     return 'mock_' + str(uuid()).lower()[:16]
 
 
-def did_name_generator(did_type='file'):
+def did_name_generator(did_type='file', name_prefix='', name_suffix='', cnt=0):
```
Comment (maany) [inline lib/rucio/tests/common.py]: @cserf could you also update the docstring, thanks  :)

File: lib/rucio/tests/temp_factories.py [hunk 3]
```diff
@@ -274,8 +275,9 @@ def _random_did(self, scope, name_prefix, name_suffix=''):
         scope = self._sanitize_or_set_scope(scope)
         if not name_prefix:
             name_prefix = 'lfn'
-        name = '%s_%s_%s%s' % (name_prefix, self.base_uuid, len(self.created_dids), name_suffix)
+        name = did_name_generator(did_type=name_prefix, name_prefix='%s_%s' % (name_prefix, self.base_uuid), name_suffix=name_suffix, cnt=len(self.created_dids))
         did = {'scope': scope, 'name': name}
+        print(did)
```
Comment (maany) [inline lib/rucio/tests/temp_factories.py]: post-debug cleanup? :P

File: lib/rucio/tests/temp_factories.py [hunk 4]
```diff
@@ -285,6 +287,7 @@ def random_did(self, scope=None, name_prefix=None, name_suffix=''):
 
     def make_dataset(self, scope=None):
         did = self._random_did(scope=scope, name_prefix='dataset')
+        print(did)
```
Comment (maany) [inline lib/rucio/tests/temp_factories.py]: same as above

File: lib/rucio/tests/common.py [hunk 5]
```diff
@@ -81,9 +84,10 @@ def did_name_generator(did_type='file'):
             return path
         path += '/file_%s' % str(uuid())
         return path
-    if did_type in ['file', 'dataset', 'container']:
-        return '%s_%s' % (did_type, str(uuid()))
-    return 'mock_' + str(uuid())
+    if name_prefix:
+        name = '%s_%s%s' % (name_prefix, cnt, name_suffix)
+        return name
+    return '%s_%s' % (did_type, str(uuid()))
```
Comment (maany) [inline lib/rucio/tests/common.py:97]: Could we add a path parameter to this method such as:
```
def  did_name_generator(path=None, did_type='file', name_prefix='', name_suffix='', cnt=0):
 if path is not None:
      return os.path.basename(path)
```
That way, we could also have some connection between a file_path and the file_name. Might save someone some debugging time in the future :P

File: lib/rucio/tests/temp_factories.py [hunk 6]
```diff
@@ -318,8 +322,9 @@ def upload_test_dataset(self, rse_name, scope=None, size=2, nb_files=2):
         self.created_dids.append(did)
         items = list()
         for _ in range(0, nb_files):
+            # TODO : Call did_name_generator
             path = file_generator(size=size)
-            name = os.path.basename(path)
```
Comment (maany) [inline lib/rucio/tests/temp_factories.py:321]: please see the comment above regarding adding the os.path.basename(path) inside the did_name_generator method :)

**Review by maany — CHANGES_REQUESTED** [review]
Thanks @cserf !! It's awesome to see the tests moving to the factories. I have left some comments, let me know what you think 👍

File: lib/rucio/tests/temp_factories.py (hunk 3 above)
Comment (cserf) [reply lib/rucio/tests/temp_factories.py]: Done

File: lib/rucio/tests/common.py [hunk 7]
```diff
@@ -66,24 +67,36 @@ def scope_name_generator():
...
+        path += 'cont_%s' % str(uuid())
         if did_type == 'container':
             return path
-        path += '/dataset_%s' % str(uuid())
+        path += 'dataset_%s' % str(uuid())
         if did_type == 'dataset':
             return path
-        path += '/file_%s' % str(uuid())
+        path += 'file_%s' % str(uuid())
         return path
```
Comment (joeldierkes) [inline lib/rucio/tests/common.py]: Handling paths by string manipulation is error-prone and hard to read, how about using the `os.path` module?

This should translate to something like:
```Python
if path is not None:
    return path

container_path = os.path.join("/belle", "mock", name_prefix, 'cont_%s' % str(uuid()))
if did_type == 'container':
    return container_path

dataset_path = os.path.join(container_path, 'dataset_%s' % str(uuid()))
if did_type == 'dataset':
    return dataset_path

file_path = os.path.join(dataset_path, 'dataset_%s' % str(uuid()))
return file_path
```

Or something like that?

**Review by joeldierkes — DISMISSED** [review]
Just a small nit, else gtg

File: lib/rucio/tests/common.py (hunk 7 above)
Comment (cserf) [reply lib/rucio/tests/common.py]: Good suggestion. Implemented.

**Comment by cserf** [comment]
Up

**Review by maany — APPROVED** [review]

**Pull request merged**